chapters/packages.tex

```diff
@@ -273,39 +267,39 @@ \subsection{External resources}\doublelabel{external-resources}
+which is counter-intuitive if \lstinline!A.B! is stored together with \lstinline!A!. When
 Modelica packages are stored in other formats a similar mapping should
 be defined, such that a resource with a given path should be unique for
 any package name that precedes it. The first part of the path may not be
 the name of a class in the package given by the authority. As above for
-"Modelica 3.2.1/package.mo" i.e. resources starting from "Modelica
-3.2.1", and "modelica://Modelica.Mechanics/C.jpg" is "Modelica
-3.2.1/Mechanics/C.jpg" - regardless of whether Modelica.Mechanics is
-stored in "Modelica 3.2.1/package.mo", "Modelica
-3.2.1/Mechanics/package.mo", or "Modelica 3.2.1/Mechanics.mo".
-
-For a Modelica-package stored as a single file, "A.mo", the resource
-"modelica://A/C.jpg" refers to a file "C.jpg" stored in the same
-directory as "A.mo", but using resources in this variant is not
+\filename{Modelica 3.2.1/package.mo} i.e. resources starting from \filename{Modelica
+3.2.1}, and \filename{modelica://Modelica.Mechanics/C.jpg} is \filename{Modelica
+3.2.1/Mechanics/C.jpg} - regardless of whether \lstinline!Modelica.Mechanics! is
+stored in \filename{Modelica 3.2.1/package.mo}, \filename{Modelica
+3.2.1/Mechanics/package.mo}, or \filename{Modelica 3.2.1/Mechanics.mo}.
+
+For a Modelica-package stored as a single file, \filename{A.mo}, the resource
+\filename{modelica://A/C.jpg} refers to a file \filename{C.jpg} stored in the same
+directory as \filename{A.mo}, but using resources in this variant is not
 recommended since multiple packages will share resources.
 
-In case the class-name contains quoted identifiers, the single-quote "`"
-and any reserved characters (``:'', ``/'', ``?'', ``\#'', ``{[}``,
-``{]}'', ``@'', ``!'', ``\$'', ``\&'', ``(``, ``)'', ``*'', ``+'',
-``,'', ``;'', ``='') should be percent-encoded as normal in URIs.
-
-{[}\emph{Example:}
+In case the class-name contains quoted identifiers, the single-quote `\lstinline!`!'
+and any reserved characters (`\lstinline!:!', `\lstinline!/!', `\lstinline!?!', `\lstinline!\#!', `\lstinline![!',
+`\lstinline!]!', `\lstinline!@!', `\lstinline!!!', `\lstinline!\$!', `\lstinline!\&!', `\lstinline!(`\lstinline!, `\lstinline!)!', `\lstinline!*!', `\lstinline!+!',
+`\lstinline!,!', `\lstinline!;!', `\lstinline!=!') should be percent-encoded as normal in URIs.
 
-\emph{Consider a top-level package} \lstinline!Modelica! \emph{and a class}
-Mechanics \emph{inside it, a reference such as}
-\lstinline!modelica://Modelica.Mechanics/C.jpg! \emph{is legal, while}
-\lstinline!modelica://Modelica/Mechanics/C.jpg! \emph{is illegal. The reference}
-\lstinline!modelica://Modelica.Mechanics/C.jpg! \emph{must also refer to a different
-resource than} \lstinline!modelica://Modelica/C.jpg!\emph{.}{]}
+\begin{example}
+Consider a top-level package \lstinline!Modelica! and a class
+\lstinline!Mechanics! inside it, a reference such as
+\lstinline!modelica://Modelica.Mechanics/C.jpg! is legal, while
```

Comment:
Or use \filename?

Reply:
Of course.  Fixed.

By the way, what about defining `\filename` to use `\mbox` to avoid hyphenation of filenames?

Reply:
Will likely need more as some of the filenames are so long that we need to break them at slashes/dots (without hyphenation, of course!).

Reply:
I don't have any experience of this.  There seems to be standard package for allowing long URLs to have line breaks (_breakurl_), that we could consider using for consistency if we also find a way to allow line breaks in `\filename`.  I wonder if this solution to a related problem could be translated to allowing line breaks after a directory separator in `\filename`:
https://www.latex4technics.com/?note=d82bg3

I guess one could also manually inject `\hspace{0pt}` at suitable places in long filenames, but it would be very nice if it could be avoided, to the point that I think I'd rather just use `\mbox` than start adding hard-to-maintain and very LaTeX specific low level commands that are just going to be annoying when trying to get any other output than PDF.

Reply:
Until we have a better solution, shouldn't we start by just applying `\mbox` so that we at least get rid of the hyphenation?

Reply:
> 
> 
> Until we have a better solution, shouldn't we start by just applying `\mbox` so that we at least get rid of the hyphenation?

Sure. As far as I understand the standard way of allowing line breaks without hyphenation is to use `\allowbreak{}` - which as far as I can see it is currently only used for long Modelica paths.

Reply:
OK, that sounds like the right tool.  Here, I found some ideas for how it can be used more conveniently (that is, without adding mess in the arguments to `\filename`):
https://gist.github.com/rauschma/d76643bab3ab7550b2dcf465e772b389

Anyway, those current examples of `\allowbreak` are a bit weird as they should be wrapped in `\lstinline`.

Reply:
Done (using `\mbox` for now).